src/components/StakingVaults/hooks/useFetchOpportunities.tsx

```diff
@@ -48,7 +46,13 @@ export const useFetchOpportunities = () => {
       knownChainIds,
     ),
     enabled:
-      !isConnected || Boolean(portfolioLoadingStatus !== 'loading' && requestedAccountIds.length),
+      // Only fetch if we're connected, not currently loading (this one is weird as we may be going back and forth between loading and idle)
+      // *and* as a last resort *if* we're not fetching Tx history, which is a bit of a hack, but it's a last resort to avoid blocking the main thread
```

Comment:
Not strictly part of this PR but might as well given our current stance on opportunities, will help later refactor. This was spewing a *lot* of XHRs on app load without a wallet, which was turbo annoying when trying to improve XHRs perfs.